### DISCUSSION TITLE
Instant mine any block

### DISCUSSION THREAD

**Discussion by zax71** [Minestom/Minestom#1690] [Help] [answered]
How do you go about instant-mining concrete with a shovel? I’m making spleef but I’d like more freedom than just snow. 

I've seen other servers do it (mineclick) but i've got no clue how to do it myself.

(1690, nice)

**Comment by GoldenCookieGIT** [comment]
Could listen to the start digging on the [Player Action Packet](https://wiki.vg/Protocol#Player_Action) and break the block that was digged. This will create delay if the player/server is lagging tho so a thing to keep in mind.

**Reply by zax71** [reply]
Thanks for that! I was looking around for events and couldn’t find one that fitted the job. I’ll mark this as accepted when I do it :)

**Reply by zax71** [reply]
Could you link to the Minestom Javadoc for that packet? I can't find it.

**Reply by dev-hydrogen** [reply]
just adding onto this, you cant recreate vanilla instamining with this as thats based on client prediction

**Reply by zax71** [reply]
> just adding onto this, you cant recreate vanilla instamining with this as thats based on client prediction

Is there a way to do it with client prediction? (Other than re-texturing a pickaxe)

**Reply by dev-hydrogen** [reply]
> > just adding onto this, you cant recreate vanilla instamining with this as thats based on client prediction
> 
> Is there a way to do it with client prediction? (Other than re-texturing a pickaxe)

efficiency 20 probably

**Comment by emortaldev** [comment, marked as answer]
you can listen for the start digging event

**Reply by zax71** [reply]
Thank you, [I used this](https://github.com/Ender-Cube/Spleef/blob/master/src/main/kotlin/me/zax71/spleef/listeners/PlayerStartDigging.kt) and it works perfectly!
